packages/netlify-cms-widget-markdown/src/serializers/remarkShortcodes.js

```diff
@@ -10,17 +10,16 @@ export function remarkParseShortcodes({ plugins }) {
 
 function createShortcodeTokenizer({ plugins }) {
   return function tokenizeShortcode(eat, value, silent) {
-    let match;
-    const potentialMatchValue = value.split('\n\n')[0].trimEnd();
-    const plugin = plugins.find(plugin => {
-      match = value.match(plugin.pattern);
-
-      if (!match) {
-        match = potentialMatchValue.match(plugin.pattern);
-      }
-
-      return !!match;
-    });
+    const matches = plugins
```

Comment:
Can this be replaced by using an `OrderedMap` in https://github.com/netlify/netlify-cms/blob/7bc6dfc39da7ce46a50a115c38ec8a721e5cf383/packages/netlify-cms-core/src/lib/registry.js#L27

Reply:
Sure! Can you say more about the benefits of using an OrderedMap here over a Map? I think either way we'll want to iterate over the full list of items, sort by match index, and then retrieve the first match.

Reply:
If we want to be consistent we need to make sure we apply the patterns always in the same order, correct?
Using an `OrderedMap` makes sure we do it always in the order components were registered via `registerEditorComponent`.

Using a match index also works, but it can be hard to predict the order in advance.

Please let me know if I'm missing anything about the need to sort by match index.

Reply:
I _think_ in this case what we care about is the order that the shortcodes appear in the markdown, as opposed to the order the components are registered. The bug was related to shortcodes being skipped due to the editorComponents Map having a different order than the appearance of the shortcodes in the markdown; even if the editorComponents are registered in an OrderedMap, they could still be ordered differently from how they appear in any given markdown string!